Eradication/VectorPopulationIndividual.cpp

```diff
@@ -818,42 +822,82 @@ namespace Kernel
         pVCI->AcquireNewInfection( &rStrain );
     }
 
-    void VectorPopulationIndividual::Vector_Migration( float dt,
-                                                       IMigrationInfo* pMigInfo,
-                                                       VectorCohortVector_t* pMigratingQueue )
+    void VectorPopulationIndividual::Vector_Migration( float dt, VectorCohortVector_t* pMigratingQueue, bool migrate_males_only )
     {
-        release_assert(pMigInfo);
+        release_assert(m_pMigrationInfoVector);
         release_assert(pMigratingQueue);
 
+        VectorPopulation::Vector_Migration(dt, pMigratingQueue, true);
+        
+        // updating migration rates and migrating females
+        IVectorSimulationContext* p_vsc = nullptr;
+        if (s_OK != m_context->GetParent()->QueryInterface(GET_IID(IVectorSimulationContext), (void**)&p_vsc))
+        {
+            throw QueryInterfaceException( __FILE__, __LINE__, __FUNCTION__, "m_context->GetParent()", "IVectorSimulationContext", "ISimulationContext" );
+        }
+        suids::suid current_node = m_context->GetSuid();
+        m_pMigrationInfoVector->UpdateRates( current_node, get_SpeciesID(), p_vsc );
+
+        Gender::Enum                            female            = Gender::FEMALE; // this is always for female vectors only, using human equivalent
+        float                                   total_rate        = m_pMigrationInfoVector->GetTotalRate( female );
+        const std::vector<float              >& r_cdf             = m_pMigrationInfoVector->GetCumulativeDistributionFunction( female );
+
+        if( ( r_cdf.size() == 0 ) || ( total_rate == 0.0 ) )
+        {
+            return; // no female vector migration
+        }
+
+        // only thing we need is for this adapter to be female; id is not used, vectors always female
+        uint32_t unused_id = 0;
+        VectorToHumanAdapter adapter( m_context, unused_id, VectorGender::VECTOR_FEMALE); 
+
+        // initializing these outside the loop, they are re-initialized just to be updated inside PickMigrationStep every time
+        suids::suid         destination = suids::nil_suid();
+        MigrationType::Enum mig_type    = MigrationType::NO_MIGRATION;
+        float               time        = 0.0;
+
         // Use the verbose "for" construct here because we may be modifying the list and need to protect the iterator.
         for( auto it = pAdultQueues->begin(); it != pAdultQueues->end(); ++it )
         {
-            IVectorCohort* tempentry = *it;
-
-            suids::suid destination = suids::nil_suid();
-            MigrationType::Enum mig_type = MigrationType::NO_MIGRATION;
-            float time = 0.0;
-            pMigInfo->PickMigrationStep( m_context->GetRng(), nullptr, 1.0, destination, mig_type, time );
+            m_pMigrationInfoVector->PickMigrationStep( m_context->GetRng(), &adapter, 1.0, destination, mig_type, time, dt );
```

Comment:
might be nice to set the vector id on the adapter.  You could have a setter method that does it.